app/operations/base_operations.rb

```diff
@@ -16,12 +16,13 @@ def initialize(model, current_user)
     def run
       ActiveRecord::Base.transaction do
         before_save
-        operate!
+        @operations = operate!
         after_save
       end
+      raise ActiveRecord::Rollback if Array(@operations).include?(false)
```

Comment:
Should we be raising this error outside a transaction? Why is it an `ActiveRecord::Rollback` instead of a more descriptive error?